### PR TITLE
ECHO REQ/RESP HANDLING

### PR DESCRIPTION
**Pull request by clstefi**
This activity deals with :
1. s11 handler for Echo Request/Response.
2. GtpV2Codec Support for the Echo Request/Response Messages.

### CODE REVIEW

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by thakurajayL** [comment]
add to whitelist

File: src/gtpV2Codec/gtpV2Stack.cpp [hunk 1]
```diff
@@ -394,6 +396,50 @@ GtpV2Stack::encodeMessage (GtpV2MessageHeader & msgHeader,
...
+                dynamic_cast<
+                EchoResponseMsg & >(msg).
+                encodeEchoResponseMsg (buffer,
+                            echoResponseStackData);
+            }
+            break;
+        }
     }
 
     Uint16 endIndex = buffer.getCurrentIndex ();
```
Comment (thakurajayL) [inline src/gtpV2Codec/gtpV2Stack.cpp:445]: -    Uint16 messageLength = (endIndex - startIndex)+8;
+    Uint16 messageLength = (endIndex - startIndex)+4; // sequence number always present
+    if(msgHeader.teidPresent) {
+        messageLength += 4; 
+    }

This is what we should have to calculate messageLength

Comment (clstefi) [reply src/gtpV2Codec/gtpV2Stack.cpp:445]: Corrected in https://github.com/omec-project/Nucleus/pull/88/commits/006fce2d116447791ab76947fa75ebbc19aca9c9

**Review by thakurajayL — APPROVED** [review]

**Pull request merged**